bin/cylc

```diff
@@ -14,643 +15,236 @@
+        exec "${COMMAND}" "--help"
+    fi
+    # Deal with cases like 'cylc --help [COMMAND/CATEGORY]'
+    if [[ (( "$#" -gt 1 )) && -f "$(ls cylc-${2}* 2>/dev/null)" ]]; then
+        local COMMAND="${CYLC_HOME_BIN}/$(ls "cylc-$2"*)"
+        exec "${COMMAND}" "--help"
+    fi
+    
+    # If category name is used, call the help func with the category
+    if [[ " ${CATEGORIES[*]} " == *" ${UTIL} "* 
+            || " ${HELP_OPTS[*]} " == *" ${UTIL} "* ]]; then
+        local COMMAND="${CYLC_HOME_BIN}/cylc-help"
+        exec "${COMMAND}" "$@"
+    fi
+    
+    # If not a category or not an actual command in the bin dir, exit
+    if [[ ! -f "$(ls "cylc-${UTIL}"*)" ]]; then
+        echo "${UTIL}: unknown utility. Abort." >&2
+        echo "Type 'cylc help' for a list of utilities."
+        return 1
+    fi
+    
+    echo "Something has gone terribly wrong if you are here..."
+    return 1 
+}
+
+category_help_command() {
+# Special case for printing categories and the gcylc gui
+    local COMMAND="${CYLC_HOME_BIN}/cylc-help"
+    exec "${COMMAND}" "categories"
+}
+
+path_lead() {
+# Ensure that ITEM_STR is at the beginning of PATH_STR
+    local PATH_STR="$1"
+    local ITEM_STR="$2"
+    if [[ -z "${PATH_STR:-}" ]]; then
+        echo "${ITEM_STR}"
+    elif [[ "${PATH_STR}" != "${ITEM_STR}" \
+            && "${PATH_STR}" != "${ITEM_STR}":* ]]; then
+        while [[ "${PATH_STR}" == *:"${ITEM_STR}" ]]; do
+            PATH_STR=${PATH_STR%:$ITEM_STR}
+        done
+        while [[ "${PATH_STR}" == *:"${ITEM_STR}":* ]]; do
+            local PATH_HEAD="${PATH_STR%:$ITEM_STR:*}"
+            local PATH_TAIL="${PATH_STR##*:$ITEM_STR:}"
+            PATH_STR="${PATH_HEAD}:${PATH_TAIL}"
+        done
+        echo "${ITEM_STR}:${PATH_STR}"
+    else
+        echo "${PATH_STR}"
+    fi
+}
+
+abort_bad_command() {
+    echo "${CYLC_NS}: ${UTIL}: unknown utility. Abort." >&2
+    echo "Type \"${CYLC_NS} help all\" for a list of utilities." >&2
+    exit 1
+}
+
+abort_ambiguous_command() {
+    echo "\"${UTIL}\" is too ambiguous to be a valid Cylc command abbreviation."
+    echo "These commands match the abbreviation \"${UTIL}\":"
+    sed -r "s,${CYLC_HOME_BIN}/cylc-,," <<<"${commands_match}"
+#    ls "${CYLC_HOME_BIN}/cylc-${UTIL}"* 2>/dev/null \
+#       | sed -r "s,${CYLC_HOME_BIN}/cylc-,,"
+    exit 1
+}
+
+run_cylc_command_matched() {
+#    COMMAND=$(ls "${CYLC_HOME_BIN}/cylc-${UTIL}"*)
+#    if [[ ! -f "${COMMAND}" || ! -x "${COMMAND}" ]]; then
+#        abort_bad_command
+#    fi
+    CYLC_UTIL="$(sed -r "s,${CYLC_HOME_BIN}/cylc-,," <<<"${commands_match}")"
+    export CYLC_UTIL
+    exec "${commands_match}" "$@"
+}
+
+run_cylc_command() {
+#    COMMAND="${commands_match}"
+#    if [[ ! -f "${COMMAND}" || ! -x "${COMMAND}" ]]; then
+#        abort_bad_command
+#    fi
+    CYLC_UTIL="${UTIL}"
+    export CYLC_UTIL
+    exec "${CYLC_HOME_BIN}/cylc-${UTIL}" "$@"
+}
+
+init_cylc
+
+UTIL="help"
+if (( "$#" > 0 )); then
```

Comment:
No need to quote `$#`. In general, it is not likely that you need to quote anything within a `(( ... ))` pair.